rcl/src/rcl/logging.c

```diff
@@ -60,7 +60,8 @@ rcl_logging_configure_with_output_handler(
   rcl_logging_output_handler_t output_handler)
 {
   RCL_CHECK_ARGUMENT_FOR_NULL(global_args, RCL_RET_INVALID_ARGUMENT);
-  RCL_CHECK_ARGUMENT_FOR_NULL(allocator, RCL_RET_INVALID_ARGUMENT);
+  RCL_CHECK_ALLOCATOR_WITH_MSG(allocator, "invalid allocator", return RCL_RET_INVALID_ARGUMENT);
```

Comment:
Why only set one of these to have an error message?

Reply:
`RCL_CHECK_ARGUMENT_FOR_NULL` has built-in error messages (see [`RCUTILS_CHECK_ARGUMENT_FOR_NULL`](https://github.com/ros2/rcutils/blob/ed1f549b02a2054597aa339cbf529ab1e2151baf/include/rcutils/error_handling.h#L172)), but  `RCL_CHECK_ALLOCATOR` doesn't.

Reply:
Sounds good

Reply:
I think that's the common scenario in the other files across `rcl`, `NULL` checks only return `RCL_RET_INVALID_ARGUMENT`